lib/util.js

```diff
@@ -48,9 +48,141 @@ function base64UrlDecode(str) {
   return atob(base64UrlToBase64(str));
 }
 
+
+function bind(fn, ctx) {
+  var additionalArgs = Array.prototype.slice.call(arguments, 2);
+  return function() {
+    var args = Array.prototype.slice.call(arguments);
+    args = additionalArgs.concat(args);
+    return fn.apply(ctx, args);
+  };
+}
+
+function isAbsoluteUrl(url) {
+  return /^(?:[a-z]+:)?\/\//i.test(url);
+}
+
+function isString(obj) {
+  return Object.prototype.toString.call(obj) === '[object String]';
+}
+
+function isObject(obj) {
+  return Object.prototype.toString.call(obj) === '[object Object]';
+}
+
+function isoToDate(str) {
+  var parts = str.match(/\d+/g),
+      isoTime = Date.UTC(parts[0], parts[1] - 1, parts[2], parts[3], parts[4], parts[5]),
+      isoDate = new Date(isoTime);
+
+  return isoDate;
+}
+
+function toQueryParams(obj) {
+  var str = [];
+  if (obj !== null) {
+    for (var key in obj) {
+      if (obj.hasOwnProperty(key) &&
+          obj[key] !== undefined &&
+          obj[key] !== null) {
+        str.push(key + '=' + encodeURIComponent(obj[key]));
+      }
+    }
+  }
+  if (str.length) {
+    return '?' + str.join('&');
+  } else {
+    return '';
+  }
+}
+
+function genRandomString(length) {
+  var randomCharset = 'abcdefghijklnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+  var random = '';
+  for (var c = 0, cl = randomCharset.length; c < length; ++c) {
+    random += randomCharset[Math.floor(Math.random() * cl)];
+  }
+  return random;
+}
+
+function extend(obj1, obj2) {
+  for (var prop in obj2) {
+    if (obj2.hasOwnProperty(prop)) {
+      obj1[prop] = obj2[prop];
+    }
+  }
+}
+
+function removeNils(obj) {
+  var cleaned = {};
+  for (var prop in obj) {
+    if (obj.hasOwnProperty(prop)) {
+      var value = obj[prop];
+      if (value !== null && value !== undefined) {
+        cleaned[prop] = value;
+      }
+    }
+  }
+  return cleaned;
+}
+
+function clone(obj) {
+  if (obj) {
+    var str = JSON.stringify(obj);
+    if (str) {
+      return JSON.parse(str);
+    }
+  }
+  return obj;
+}
+
+// Analogous to _.omit
+function omit(obj) {
+  var props = Array.prototype.slice.call(arguments, 1);
+  var newobj = {};
+  for (var p in obj) {
+    if (obj.hasOwnProperty(p) && props.indexOf(p) == -1) {
+      newobj[p] = obj[p];
+    }
+  }
+  return clone(newobj);
+}
+
+function find(collection, searchParams) {
+  var c = collection.length;
+  while (c--) {
+    var item = collection[c];
+    var found = true;
+    for (var prop in searchParams) {
+      if (!searchParams.hasOwnProperty(prop)) {
+        continue;
+      }
+      if (item[prop] !== searchParams[prop]) {
+        found = false;
+        break;
+      }
+    }
+    if (found) {
+      return item;
+    }
+  }
+}
+
 module.exports = {
   base64UrlToBase64: base64UrlToBase64,
   base64UrlToString: base64UrlToString,
   stringToBuffer: stringToBuffer,
-  base64UrlDecode: base64UrlDecode
+  base64UrlDecode: base64UrlDecode,
```

Comment:
Sounds good. It might justify pulling these base64 functions out into their own npm module